lib/sshutils/sftp/sftp.go

```diff
@@ -63,9 +61,9 @@ type Config struct {
 	// SSH session
 	getHomeDir homeDirRetriever
 
-	// ProgressWriter is a writer for printing the progress
+	// ProgressWriter is a callback to return writer for printing the progress
 	// (used only on the client)
-	ProgressWriter io.Writer
+	ProgressWriter func(fileInfo os.FileInfo) io.Writer
```

Comment:
Tip for the future: the previous commit depends on the changes introduced in this commit. It'd be easier to understand if this commit was first followed by the other.